File: src/services/gateway/src/main/java/io/kuz/ecom/gateway/common/mapper/PaginationMetaMapper.java
package io.kuz.ecom.gateway.common.mapper;

import io.kuz.ecom.gateway.common.dto.PaginationMetaDTO;
import io.kuz.ecom.proto.common.PaginationMeta;

public class PaginationMetaMapper {

    public static PaginationMetaDTO toDTO(PaginationMeta proto) {
        return new PaginationMetaDTO(
            (long) proto.getPage(),
            (long) proto.getPageSize(),
            (long) proto.getPageCount()
        );
    }
}

File: grpc-mapper/src/main/kotlin/io/kuz/grpc/mapper/common/PaginationMetaMapper.kt
package io.kuz.grpc.mapper.common

import io.kuz.ecom.common.models.common.PaginationMetaModel
import io.kuz.ecom.proto.common.PaginationMeta

object PaginationMetaMapper {

    fun toGRPC(model: PaginationMetaModel): PaginationMeta {
        return PaginationMeta
            .newBuilder()
            .setOffset(model.offset.toInt())
            .setLimit(model.limit.toInt())
            .setTotal(model.total.toInt())
            .build()
    }

    fun toModel(grpc: PaginationMeta): PaginationMetaModel {
        return  PaginationMetaModel(
            limit = grpc.limit.toLong(),
            offset = grpc.offset.toLong(),
            total = grpc.total.toLong(),
        )
    }
}
